as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/tools/as/core/server/controllable/subsystems/internal/WildFlyFileSystemPublishController.java

```diff
@@ -0,0 +1,63 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.as.core.server.controllable.subsystems.internal;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.ide.eclipse.as.core.server.IModulePathFilter;
+import org.jboss.ide.eclipse.as.core.server.internal.v7.DeploymentMarkerUtils;
+import org.jboss.ide.eclipse.as.wtp.core.server.behavior.LocalFilesystemController;
+
+public class WildFlyFileSystemPublishController extends StandardFileSystemPublishController {
+	protected int removeModule(IModule[] module, IPath remote, IProgressMonitor monitor) throws CoreException {
+		undeployModuleAndWait(module, remote, 15000, monitor);
+		return super.removeModule(module, remote, monitor);
+	}
+	protected IStatus[] executeFullPublish(IModule[] module, IPath archiveDestination, IModulePathFilter filter, IProgressMonitor monitor) 
+			throws CoreException {
+		undeployModuleAndWait(module, archiveDestination, 15000, monitor);
```

Comment:
should we proceed if undeployModuleAndWait return false? at the very least we should log the failed un-deployment attempt